### ISSUE TITLE
Use GPU and multi-card for model training

### ISSUE THREAD

**Issue by yumianhuli2** [ZikangZhou/HiVT#3]
Hello! How to use GPU and multi-card for training? The default card 0 is the CPU for training.
Thank U!

**Comment by ZikangZhou** [comment]
This repo uses pytorch-lightning as the trainer. It's convenient to do multi-gpu training by:

```
python train.py --root /path/to/dataset_root/ --embed_dim 128 --gpus #YOUR_GPU_NUM
```

If I remember correctly, this will use Pytorch DDP Spawn strategy by default. If you want to use Pytorch DDP instead (which should be faster than DDP Spawn in general), you can add one line to **train.py**:

```
parser.add_argument('--strategy', type=str, default='ddp')
```

Let me know if it works.

**Comment by ZikangZhou** [comment]
@yumianhuli2 To reproduce the results in the paper when using multi-gpu training, please also make sure that the effective batch size (batch_size * gpu_num) is 32. For example, if you use 4 gpus, then the batch size per gpu should be 8:

```
python train.py --root /path/to/dataset_root/ --embed_dim 128 --gpus 4 --train_batch_size 8
```

**Comment by yumianhuli2** [comment]
Thank you！

**Comment by tandangzuoren** [comment]
> 

Thank you for your outstanding work！If the batchsize is changed, does the learning rate need to be adjusted accordingly？

**Comment by ZikangZhou** [comment]
@tandangzuoren I believe the learning rate should be adjusted. The number of epochs may also need to be changed.

**Comment by tteokl** [comment]
@ZikangZhou Thank you for your advice on this. May I know why 32 is the effective batch size?
